Destination check

JakobRoseke opened the pull request:
This method should work, however, it is not perfect. As I mentioned in one of the commits, it COULD in theory miss that is has reached the destination. There are obviously workarounds for that but we could discuss if we want to keep this feature or not.

JakobRoseke committed:
Added first try at self destination checking
It works by having two Int that together become car.getDistance(), one is correct value and the other is wrong. So when the correct value is == distanceToGo and incorrect value you are at the destination

JakobRoseke committed:
Added one more variable and initialized the others

JakobRoseke committed:
changed the behaviour of the function
I changed what happens when it reached the destination to change the CURRENT_SPEED to 0 and set car to it. Also made the distanceToDrive equal the originalDistance plus the distance the car has driven in the wrong direction.

JakobRoseke committed:
Changed the gyroscope offset to 2
I calibrated the gyroscope with the example code and got 2 as value

JakobRoseke committed:
removed unwanted code from the checkDestination
I tested this and it works relatively well, the only thing is that since it loops VERY quickly there might a change that in doesn't recognize that you have driven the distanceToDrive. It could be changed to a => instead of ==

JakobRoseke committed:
Added forgotten pinMode in setup
> We forgot to add the pinMode in the setup for the automation mode, I added this in this branch because I noticed it here and felt like it wasn't a major change for its own branch.

Ali-karkhaneh reviewed (approved):
has conflicts, which lines are not needed?

Ali-karkhaneh reviewed (changes requested):
resolve conflicts, which lines are needed?

JakobRoseke committed:
Merge branch 'master' into destinationCheck

JakobRoseke commented:
> resolve conflicts, which lines are needed?

There, I added the lines wanted.

JakobRoseke committed:
removed variable declared twice

accidentally left the variable in the merge conflict resolve, I removed it now.
